### PR TITLE
update space quota on upload, delete and restore

### PR DESCRIPTION
**Pull request by JanAckermann**
## Description
**Note: While uploading files with uppy we check the quota of the respective space, this is why we should keep spaces in sync**

Tackling the following issues:

- Uploading or deleting files doesn't update project spaces quota
- Restoring files doesn't update project spaces or personal quota 
- Reload outside of the personal home doesn't show quota in the user menu

## Related Issue

- Fixes https://github.com/owncloud/web/issues/6930
- Fixes https://github.com/owncloud/web/issues/7389

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/27400/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27400/tracing/participant-management-alice-2022-8-8-11-00-45.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27400/tracing/participant-management-anonymous-2022-8-8-11-01-01.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27400/tracing/participant-management-brian-2022-8-8-11-00-50.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27400/tracing/participant-management-carol-2022-8-8-11-00-55.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27400/tracing/public-link-alice-2022-8-8-10-58-33.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27400/tracing/public-link-anonymous-2022-8-8-10-58-46.zip 


</p></details>

File: packages/web-runtime/src/store/user.ts [hunk 1]
```diff
@@ -46,6 +46,10 @@ const mutations = {
     state.language = user.language
     state.role = user.role
     sentrySetUser({ username: user.id })
+
+    if (user.quota) {
+      this.commit('SET_QUOTA', user.quota)
+    }
```
Comment (JanAckermann) [inline packages/web-runtime/src/store/user.ts:52]: Fixes problem: Navigate to spaces -> Reload page -> Quota not shown in user menu

File: packages/web-app-files/src/components/SpaceQuota.vue [hunk 2]
```diff
@@ -45,7 +45,7 @@ export default {
       return filesize(this.spaceQuota.used)
     },
     quotaUsagePercent() {
-      return ((this.spaceQuota.used / this.spaceQuota.total) * 100).toFixed(1)
+      return parseFloat(((this.spaceQuota.used / this.spaceQuota.total) * 100).toFixed(2))
```
Comment (JanAckermann) [inline packages/web-app-files/src/components/SpaceQuota.vue:48]: Aligned to oc10 user quota view

File: packages/web-runtime/src/services/auth/userManager.ts [hunk 3]
```diff
@@ -212,12 +212,13 @@ export class UserManager extends OidcUserManager {
       email: login?.email || user?.email || '',
       groups: userGroups,
       role,
-      language: login?.language,
-      ...(user.quota &&
-        user.quota.definition !== 'default' &&
-        user.quota.definition !== 'none' && { quota: user.quota })
```
Comment (JanAckermann) [inline packages/web-runtime/src/services/auth/userManager.ts:218]: I don't get this conditions, we load the user quota on other places without those conditions

File: packages/web-app-files/src/components/AppBar/CreateAndUpload.vue [hunk 4]
```diff
@@ -278,23 +281,32 @@ export default defineComponent({
...
 
-    onFileSuccess() {
-      if (this.user?.quota) {
-        this.SET_QUOTA(this.user.quota)
-      }
-    },
-
-    onUploadComplete(result) {
+    async onUploadComplete(result) {
+      console.log('COMPLETE')
```
Comment (JanAckermann) [inline packages/web-app-files/src/components/AppBar/CreateAndUpload.vue:288]: OOOPS

File: packages/web-app-files/src/mixins/deleteResources.js [hunk 5]
```diff
@@ -160,9 +163,27 @@ export default {
         this.toggleModalConfirmButton()
 
         // Load quota
-        if (this.user?.id) {
```
Comment (JanAckermann) [inline packages/web-app-files/src/mixins/deleteResources.js:163]: User should be always there, that's why I omit this condition

File: packages/web-runtime/src/services/auth/userManager.ts (hunk 3 above)
Comment (kulmann) [reply packages/web-runtime/src/services/auth/userManager.ts:218]: Me neither 😞

File: packages/web-runtime/src/store/user.ts [hunk 6]
```diff
@@ -51,6 +51,12 @@ const mutations = {
     state.capabilities = data.capabilities
     state.version = data.version
   },
+  /**
+   * Legacy for oC10, in OCIS quota sticks on the respective drive
```
Comment (kulmann) [inline packages/web-runtime/src/store/user.ts:55]: FYI was thinking with @dschmidt today that we might want to introduce a virtual personal drive for oc10 compatibility... nothing for this PR though

**Review by kulmann — COMMENTED** [review]
Setting the quota in spaces / non-spaces environments has quite some code duplication across the different files touched in this PR. Do you want to continue to work on that or keep it as is for now?

File: packages/web-runtime/src/store/user.ts (hunk 6 above)
Comment (JanAckermann) [reply packages/web-runtime/src/store/user.ts:55]: Would be nice 💪😊

**Comment by JanAckermann** [comment]
@kulmann If you can give me some tips, I can for sure remove some code duplication

File: packages/web-runtime/src/components/Topbar/UserMenu.vue [hunk 7]
```diff
@@ -116,14 +133,24 @@ export default {
       )
     },
     limitedPersonalStorage() {
+      if (!this.useLegacyQuota) {
+        return this.quota.total !== 0
```
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/UserMenu.vue:137]: Evaluates to `true` if the `total` field is absent. Afaik this should evaluate to `false` then.

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-runtime/src/components/Topbar/UserMenu.vue (hunk 7 above)
Comment (JanAckermann) [reply packages/web-runtime/src/components/Topbar/UserMenu.vue:137]: Total 0 means unlimited in the ocis world

File: packages/web-app-files/src/mixins/deleteResources.js (hunk 5 above)
Comment (kulmann) [reply packages/web-app-files/src/mixins/deleteResources.js:163]: No, deleting resources is possible on public links with edit permissions.

File: packages/web-runtime/src/components/Topbar/UserMenu.vue (hunk 7 above)
Comment (kulmann) [reply packages/web-runtime/src/components/Topbar/UserMenu.vue:137]: But if `total` is not specified I'd interpret it as the same as `0`. Needs to be clarified then I guess. 😅

File: packages/web-app-files/src/mixins/deleteResources.js (hunk 5 above)
Comment (JanAckermann) [reply packages/web-app-files/src/mixins/deleteResources.js:163]: But I have a location check here ?

Comment (kulmann) [reply packages/web-app-files/src/mixins/deleteResources.js:163]: lol, nevermind. Sorry, location check was not visible on the front page of the PR where I stumbled over this comment.

File: packages/web-runtime/src/components/Topbar/UserMenu.vue (hunk 7 above)
Comment (JanAckermann) [reply packages/web-runtime/src/components/Topbar/UserMenu.vue:137]: Yes you are right if 0 or not specified, means unlimited, was just a note for me

**Comment by kulmann** [comment]
> @kulmann If you can give me some tips, I can for sure remove some code duplication

Nevermind. Just saw that the code duplication is in mixins. As we want to get rid of those it's not worth it I think.

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Pull request merged**